backup/restore.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+this_script="/opt/backup/restore.sh"
+
+set -euo pipefail
+
+backup_file="${1:-}"
+
+if [ ! -f "${backup_file}" ] ; then
+    echo "ERROR: missing backup tarball file."
+    echo "Usage: $0 <backup-tarball>"
+    echo "                  <backup-tarball> - backup created with 'backup.sh'"
+    echo
+    exit 1
+fi
+
+# Only sith lords and backup files deal in absolutes.
+backup_file="$(realpath "${backup_file}")"
+
+if    [[ "${backup_file}" == /opt/mastodon/* ]] \
+   || [[ "${backup_file}" == /opt/backup/* ]] \
+   || [[ "${backup_file}" == /opt/caddy/* ]] ; then
+   echo "ERROR: backup file '${backup_file}' resides in a path that wiĺl be WIPED!"
+   echo "       Move outside /opt/mastodon, /opt/backup, or /opt/caddy and try again."
+   echo
+   exit 1
+fi
+
+
+reinit_script="/opt/backup/re_initialise.sh"
+
+if [ "$(realpath "$0")" = "$this_script" ] ; then
+    # The wiping below will also remove this very script, so we create a tempfile to run from.
+    tmp_restore_script="$(mktemp)"
+    chmod 700 "${tmp_restore_script}"
+    cat "${this_script}" >> "${tmp_restore_script}"
```

Comment:
Any reason for not using `cp -a "${this_script}" "${tmp_restore_script}"`?

Reply:
I'd like the script to take the attributes of the temp file (0700) instead of preserving its own.

Reply:
Fair enough. My question now is whether the wiping of the running script is even a problem. I did an experiment.

```
$ cat ./test.sh
#!/bin/bash

rm -f "${0}"

for i in {1..3}; do
	echo "sleep nr ${i}"
	sleep 1
done
$ cp -a test.sh test2.sh
$ ./test2.sh
sleep nr 1
sleep nr 2
sleep nr 3
$ ./test2.sh 
bash: ./test2.sh: No such file or directory
```

So this probably shouldn't be necessary, right?